scipy/sparse/linalg/_isolve/tests/test_iterative.py

```diff
@@ -478,29 +472,35 @@ def test_maxiter_worsening(solver):
                   [0.1112795288033368, 0j, 0j, -0.16127952880333785]])
     v = np.ones(4)
     best_error = np.inf
-    tol = 7 if platform.machine() == 'aarch64' else 5
+
+    # Unable to match the Fortran code tolerance levels with this example
+    # Original tolerance values
+
+    # slack_tol = 7 if platform.machine() == 'aarch64' else 5
+    slack_tol = 9
 
     for maxiter in range(1, 20):
         x, info = solver(A, v, maxiter=maxiter, tol=1e-8, atol=0)
 
         if info == 0:
-            assert np.linalg.norm(A @ x - v) <= 1e-8 * np.linalg.norm(v)
+            assert norm(A @ x - v) <= 1e-8 * norm(v)
 
         error = np.linalg.norm(A @ x - v)
         best_error = min(best_error, error)
 
         # Check with slack
-        assert error <= tol * best_error
+        assert error <= slack_tol * best_error
 
 
 def test_x0_working(solver):
     # Easy problem
-    np.random.seed(1)
+    # np.random.seed(1)
+    rng = np.random.default_rng(1)
```

Comment:
Can we use a larger seed here and remove the commented out old code?

Reply:
Oh it's a brainfart. Also forgot to modify the TestGMRES random parts. I'll fix an ARPACK test first then fix these. Thanks.